pkg/reconciler/delivery/policy.go

```diff
@@ -64,6 +64,10 @@ func translatePolicy(p *v1alpha1.Policy) *Policy {
 		}
 		stages[i] = Stage{p.Spec.Stages[i].Percent, thresholdptr}
 	}
+	// prepend a 0-stage if first stage has non-zero or doesn't exist
+	if len(stages) == 0 || stages[0].Percent != 0 {
```

Comment:
Is it possible to make the rollout algorithm work without the 0 state in the stages list rather than mutating the policy object in memory?

Reply:
I agree that mutating the `v1alpha1.Policy` object is not good because that's the custom resource that persists in K8s. However, here I'm translating `v1alpha1.Policy` to `delivery.Policy` which is only an internally-used one-time thing. This doesn't mutate `v1alpha1.Policy`.

Plus, although the rollout algorithm can work well without the 0 state, adding a 0-state can make all the policy-related helper functions much neater.